File: code/generate_strategies/IR(CoT)/python/python/function_Pydantic_conint.py
from pydantic import BaseModel, PositiveFloat, ValidationError
from typing import Annotated

class ProductValidator:
    def create_product(self, stock: int, price: float) -> str:
        StockType = Annotated[int, 0 <= 1000] 

        class Product(BaseModel):
            stock: Annotated[int, 0 <= 1000]
            price: PositiveFloat 

        try:
            product = Product(stock=stock, price=price)
            return str(product)
        except ValidationError as e:
            return str(e.errors())

File: code/generate_strategies/IR(CoT)/cpp/cpp/function_Pydantic_conint.cpp

#include <string>
#include <nlohmann/json.hpp>
#include <sstream>
#include <vector>
#include <stdexcept>
class ProductValidator {
public:
    std::string create_product(int stock, float price) {
        
        nlohmann::json errors = nlohmann::json::array();
        bool has_errors = false;
    
        
        if (stock < 0 || stock > 1000) {
            has_errors = true;
            nlohmann::json error;
            error["loc"] = {"stock"};
            error["msg"] = "value is not within range [0, 1000]";
            error["type"] = "value_error.range";
            errors.push_back(error);
        }
    
        
        if (price <= 0) {
            has_errors = true;
            nlohmann::json error;
            error["loc"] = {"price"};
            error["msg"] = "ensure this value is greater than 0";
            error["type"] = "value_error.number.not_gt";
            errors.push_back(error);
        }
    
        if (has_errors) {
            return errors.dump();
        }
    
        
        std::ostringstream result;
        result << "stock=" << stock << " price=" << price;
        return result.str();
    }
};